src/components/page/elements/QuizElement.js

```diff
@@ -1,35 +1,51 @@
 import React from 'react'
 import MultipleChoiceQuestionElement from './MultipleChoiceQuestionElement'
+import { checkQuizStatus, markQuizSolved } from '../../../actions/quiz'
+import { connect } from 'react-redux'
 
 class QuizElement extends React.Component {
   constructor(props) {
     super(props)
     this.state = { currentQuestion: 0 }
     this.nextQuestionClicked = this.nextQuestionClicked.bind(this)
+    this.resetClicked = this.resetClicked.bind(this)
   }
 
   componentDidMount() {
+    if (!this.quizCompleted()) {
+      this.props.checkQuizStatus(this.props.element.id)
+    }
     if (this.props.content) {
       this.setState({ currentQuestion: 0, maxQuestions: this.props.content.length, finished: false })
     }
   }
 
   nextQuestionClicked() {
     const nextQuestion = this.state.currentQuestion + 1
-    if (nextQuestion === this.state.maxQuestions) return this.setState({ finished: true })
+    if (nextQuestion === this.state.maxQuestions) {
+      this.props.markQuizSolved(this.props.element.id)
+      return this.setState({ finished: true })
+    }
     return this.setState({ currentQuestion: nextQuestion })
   }
 
+  resetClicked() {
+    this.setState({ finished: false, currentQuestion: 0 })
+  }
+
   render() {
     return (
       <div className={'hero mb30 has-background-info exercise-box'}>
         <div className={'exercise-header'}>
-          <p className={'subtitle has-text-white has-text-weight-bold is-marginless'}>
-            Quiz
-            <span className={'is-pulled-right is-right has-text-left'}>
+          <div className={'subtitle has-text-white has-text-weight-bold is-marginless'}>
+            <div>
```

Comment:
Is this `div` useful?

Reply:
yes, it's necessary for proper alignment because of the flexbox